models/concerns/searchable.rb

```diff
@@ -25,30 +25,40 @@ def as_indexed_json(options={})
     # code runs single-threaded.
     @@enable_es = true
 
-    # The caller may opt to temporarily disable ES callbacks by using this toggle, e.g.:
-    #
-    #   comment.enable_es(false)
-    #   comment.update!(data)
-    #   comment.enable_es(true)
-    #
-    def enable_es(x)
-      @@enable_es = x
+    def es_enabled?
+      @@enable_es
     end
 
-    def is_es_enabled
-      @@enable_es
+    def without_es
+      # A "Context Manager" to temporarily disable elasticsearch callbacks.  Whatever happens, this makes
+      # sure that enable_es is restored.  E.g.:
+      #
+      #   comment.without_es do
+      #     comment.update!(data)
+      #   end
+      #
+      original_enable_es = es_enabled?
+      @@enable_es = false
+      begin
+        yield
```

Comment:
Ah - Ruby has `yield`!

Reply:
Yep! And remarkably, it behaves the same way too.